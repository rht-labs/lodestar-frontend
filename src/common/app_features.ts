// All roles defined by the backend to be used in the frontend should be enumerated in this object.

export const APP_FEATURES = {
  admin: 'admin',
  writer: 'writer',
  reader: 'reader',
  notifications: 'notifications',
  engagementCardIcons: 'engagementCardIcons',
  resetUser: 'resetUser',
};

export type AppFeature = keyof typeof APP_FEATURES;
