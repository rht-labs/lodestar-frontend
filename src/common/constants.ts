export const APPLICATION_CONTENT_CONTAINER_ID = 'main-application-content';
