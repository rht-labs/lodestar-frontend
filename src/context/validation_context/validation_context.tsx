import React, { useState, useCallback } from 'react';
import { Validator } from '../../schemas/validators/validator';

export interface ValidationContext {
  validate: (fieldName: string) => (value: any) => string[];
  getValidationResult: (fieldName: string) => string[];
  validationResults: { [key: string]: any };
  clearValidationResults: () => void;
}

export type FormValidator = { [key: string]: Validator[] };

export const ValidationContext = React.createContext<ValidationContext>({
  validate: (fieldName: string) => (value: any) => [],
  getValidationResult: (fieldName: string) => [],
  clearValidationResults: () => {},
  validationResults: {},
});

const { Provider } = ValidationContext;

export const ValidationProvider = ({
  children,
  validators = {},
}: {
  children?: any;
  validators: FormValidator;
}) => {
  const [validationResults, setValidationResults] = useState({});

  const getValidationResult = (fieldName: string) => {
    if (validationResults && validationResults[fieldName]) {
      return validationResults[fieldName];
    }
    return [];
  };

  const clearValidationResults = useCallback(() => {
    setValidationResults({});
  }, [setValidationResults]);

  const validate = useCallback(
    (fieldName: string) => (value: any) => {
      if (validators && validators[fieldName]) {
        const result = validators[fieldName]
          .reduce(
            (accumulatedErrors, currentValidator) => [
              ...accumulatedErrors,
              currentValidator?.(value),
            ],
            []
          )
          .filter(message => !!message);
        setValidationResults({
          ...validationResults,
          [fieldName]: result,
        });
        return result;
      } else {
        return [];
      }
    },
    [validators, validationResults]
  );

  return (
    <Provider
      value={{
        validationResults,
        getValidationResult,
        clearValidationResults,
        validate,
      }}
    >
      {children}
    </Provider>
  );
};
