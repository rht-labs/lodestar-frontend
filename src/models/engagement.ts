export interface Engagement {
  name: string;
}
export class Engagement {
  constructor({ name }: Engagement) {
    this.name = name;
  }
  name: string;
}
