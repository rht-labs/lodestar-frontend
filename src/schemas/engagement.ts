import faker from 'faker/locale/en_US';
import { LaunchData } from './launch_data';
import { GitCommit } from './git_commit';
import { CreationDetails } from './creation_details';
import { ClusterStatus } from './cluster_status';
export interface Engagement {
  additional_details?: string;
  archive_date: Date;
  commits: GitCommit[];
  customer_contact_email: string;
  customer_contact_name: string;
  customer_name: string;
  description: string;
  end_date: Date;
  engagement_lead_email: string;
  engagement_lead_name: string;
  engagement_users: Array<any>;
  last_update: string;
  location: string;
  mongo_id?: string;
  ocp_cloud_provider_name: string;
  ocp_cloud_provider_region: string;
  ocp_cluster_size: string;
  ocp_persistent_storage_size: string;
  ocp_sub_domain: string;
  ocp_version: string;
  project_id: number;
  project_name: string;
  start_date: Date;
  technical_lead_email: string;
  technical_lead_name: string;
  launch?: LaunchData;
  creation_details: CreationDetails;
  last_update_by_name: string;
  suggested_subdomain?: string;
  status: ClusterStatus;
}
export abstract class Engagement {
  static fromFake(staticData = false): Engagement {
    return {
      additional_details: staticData
        ? 'Additional information here'
        : faker.lorem.paragraphs(2),
      archive_date: staticData ? new Date(2020, 6, 30) : faker.date.recent(),
      commits: [],
      customer_contact_email: staticData
        ? 'bob@doe.com'
        : faker.internet.email(),
      customer_contact_name: staticData
        ? 'Bob Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      customer_name: staticData ? 'NASA' : faker.company.companyName(),
      description: staticData
        ? "It's rocket science"
        : faker.lorem.paragraphs(5),
      end_date: staticData ? new Date(2020, 6, 1) : faker.date.future(),
      engagement_users: staticData ? [] : [],
      engagement_lead_email: staticData
        ? 'alice@doe.com'
        : faker.internet.email(),
      engagement_lead_name: staticData
        ? 'Alice Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      last_update: staticData ? '2020-01-01' : new Date().toString(),
      location: staticData
        ? 'Nashville, TN'
        : `${faker.address.city()}, ${faker.address.stateAbbr()}`,
      mongo_id: '1',
      ocp_cloud_provider_name: staticData ? 'AWS' : 'AWS',
      ocp_cloud_provider_region: staticData
        ? 'N. Virginia'
        : faker.lorem.word(),
      ocp_cluster_size: staticData ? 'Large' : 'Large',
      ocp_persistent_storage_size: staticData
        ? '100G'
        : `${faker.random.number()}G`,
      ocp_sub_domain: staticData ? 'ordoabchao' : faker.lorem.word(),
      ocp_version: staticData ? '4.4' : faker.random.number().toString(),
      project_id: staticData ? 1 : faker.random.number(),
      project_name: staticData ? 'Boots on the Moon' : faker.company.bsNoun(),
      start_date: staticData ? new Date(2020, 5, 1) : faker.date.recent(),
      technical_lead_email: staticData ? 'eve@doe.com' : faker.internet.email(),
      technical_lead_name: staticData
        ? 'Eve Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      creation_details: staticData
        ? {
            created_by_email: 'dwasinger@redhat.com',
            created_by_user: 'dwasinge',
            created_on: new Date(2020, 1, 1),
          }
        : {
            created_by_email: faker.internet.email(),
            created_by_user: `${faker.name.firstName()} ${faker.name.lastName()}`,
            created_on: faker.date.recent(),
          },
      last_update_by_name: staticData
        ? 'James Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      launch: staticData
        ? null
        : faker.random.boolean()
        ? {
            launched_by: faker.name.firstName(),
            launched_date_time: faker.date.recent(),
          }
        : null,
      status: ClusterStatus.fromFake(staticData),
    };
  }
}

export enum EngagementStatus {
  active = 'active',
  past = 'past',
  upcoming = 'upcoming',
}

export const getEngagementStatus = (
  engagement: Engagement
): EngagementStatus => {
  if (!engagement) {
    return null;
  }
  const Today = new Date();
  const { launch, end_date } = engagement;
  const hasLaunched = !!launch?.launched_date_time;
  if (hasLaunched && end_date >= Today) {
    return EngagementStatus.active;
  } else if (hasLaunched && end_date < Today) {
    return EngagementStatus.past;
  } else {
    return EngagementStatus.upcoming;
  }
};
