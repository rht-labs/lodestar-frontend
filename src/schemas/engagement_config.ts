import FakedSchema from './fixtures/engagement_form_config.json';

export interface EngagementFormOption {
  label: string;
  value: string;
  options?: EngagementFormOption[];
}

export interface EngagementFormValidator {
  kind: string;
  value?: any;
}

export interface FormConfig {
  options?: EngagementFormOption[];
  validators?: EngagementFormValidator[];
}

export interface EngagementFormConfig {
  [key: string]: { [key: string]: FormConfig };
}

export abstract class EngagementFormConfig {
  static fromFake(): EngagementFormConfig {
    return FakedSchema;
  }
}
