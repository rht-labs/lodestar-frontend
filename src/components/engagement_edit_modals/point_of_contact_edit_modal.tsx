import React from 'react';
import { Engagement } from '../../schemas/engagement_schema';
import {
  Modal,
  ModalVariant,
  Button,
  TextInput,
  InputGroupText,
  InputGroup,
  FormGroup,
  Form,
} from '@patternfly/react-core';
import { useModalVisibility } from '../../context/edit_modal_visibility_context/edit_modal_visibility_hook';
import { EditModalTemplate } from '../../layout/edit_modal_template';
import { UserIcon, EnvelopeIcon } from '@patternfly/react-icons';
import { useFeatures } from '../../context/feature_toggles/feature_hook';
import { APP_FEATURES } from '../../common/app_features';
import { useValidation } from '../../context/validation_context/validation_hook';
export interface PointOfContactEditModalProps {
  onChange: (fieldName: string, value: any) => void;
  formOptions: object;
  engagement: Engagement;
  isOpen: boolean;
  onSave: (engagement: Engagement) => void;
}
export function PointOfContactEditModal({
  onChange,
  engagement,
  isOpen,
  onSave: propsOnSave,
}: PointOfContactEditModalProps) {
  const { requestClose } = useModalVisibility();
  const { hasFeature } = useFeatures();

  const input: React.CSSProperties = {
    backgroundColor: '#EDEDED',
  };
  const { validate, getValidationResult } = useValidation();

  const onSave = () => {
    propsOnSave(engagement);
    requestClose();
  };

  return (
    <Modal
      variant={ModalVariant.small}
      isOpen={isOpen}
      onClose={requestClose}
      title="Points of Contact"
    >
      <EditModalTemplate
        actions={
          <div>
            <Button onClick={onSave} data-cy={'save_point_of_contact'}>
              Save
            </Button>
          </div>
        }
      >
        <Form>
          <FormGroup
            fieldId="engagement_lead"
            helperText="Who is the Engagement Lead?"
            isRequired
            label="Labs EL"
            helperTextInvalid={getValidationResult(
              'engagement_lead_email'
            ).join(' ')}
            validated={
              getValidationResult('engagement_lead_email').length > 0
                ? 'error'
                : 'default'
            }
          >
            <InputGroup>
              <InputGroupText
                style={input}
                component="label"
                htmlFor="engagement-lead"
              >
                <UserIcon />
              </InputGroupText>
              <TextInput
                aria-label="Engagement lead name"
                style={input}
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                id="name"
                name="engagement-lead-name"
                onChange={e => {
                  onChange('engagement_lead_name', e);
                }}
                placeholder="Full Name"
                type="text"
                value={engagement.engagement_lead_name || ''}
                data-cy={'engagement_lead_name'}
              />
              <InputGroupText
                style={input}
                component="label"
                htmlFor="engagement-lead-email"
              >
                <EnvelopeIcon />
              </InputGroupText>
              <TextInput
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                aria-label="engagement lead email"
                id="email"
                style={input}
                name="engagement-lead-email"
                onChange={e =>
                  validate('engagement_lead_email')(e) &&
                  onChange('engagement_lead_email', e)
                }
                placeholder="Email Address"
                type="email"
                value={engagement.engagement_lead_email || ''}
                data-cy={'engagement_lead_email'}
              />
            </InputGroup>
          </FormGroup>

          <FormGroup
            fieldId="technical_lead"
            isRequired
            helperText="Who is the Tech Lead?"
            label="Labs Technical Lead"
            helperTextInvalid={getValidationResult('technical_lead_email').join(
              ' '
            )}
            validated={
              getValidationResult('technical_lead_email').length > 0
                ? 'error'
                : 'default'
            }
          >
            <InputGroup label="Labs Tech Lead">
              <InputGroupText
                style={input}
                component="label"
                htmlFor="tech-lead-name"
              >
                <UserIcon />
              </InputGroupText>
              <TextInput
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                aria-label="Tech lead name"
                id="tech-lead-name"
                style={input}
                name="tech-lead-name"
                onChange={e => onChange('technical_lead_name', e)}
                placeholder="Full Name"
                type="text"
                value={engagement.technical_lead_name || ''}
                data-cy={'tech_lead_name'}
              />
              <InputGroupText
                style={input}
                component="label"
                htmlFor="tech-lead-email"
              >
                <EnvelopeIcon />
              </InputGroupText>
              <TextInput
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                style={input}
                aria-label="tech lead email"
                id="tech-lead-email"
                name="tech-lead-email"
                onChange={e =>
                  validate('technical_lead_email')(e) &&
                  onChange('technical_lead_email', e)
                }
                placeholder="Email Address"
                type="email"
                value={engagement.technical_lead_email || ''}
                data-cy={'tech_lead_email'}
              />
            </InputGroup>
          </FormGroup>

          <FormGroup
            fieldId="customer-contact"
            helperText="Who is the point person for the project?"
            isRequired
            label="Customer Contact"
            helperTextInvalid={getValidationResult(
              'customer_contact_email'
            ).join(' ')}
            validated={
              getValidationResult('customer_contact_email').length > 0
                ? 'error'
                : 'default'
            }
          >
            <InputGroup label="Customer Contact">
              <InputGroupText
                style={input}
                component="label"
                htmlFor="customer-contact-name"
              >
                <UserIcon />
              </InputGroupText>
              <TextInput
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                style={input}
                aria-label="Customer contact name"
                id="customer-contact-name"
                name="customer-contact-name"
                onChange={e => onChange('customer_contact_name', e)}
                type="text"
                placeholder="Full Name"
                value={engagement.customer_contact_name || ''}
                data-cy={'customer_contact_name'}
              />
              <InputGroupText
                style={input}
                component="label"
                htmlFor="customer-contact-email"
              >
                <EnvelopeIcon />
              </InputGroupText>
              <TextInput
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                aria-label="customer contact email"
                id="customer-contact-email"
                name="customer-contact-email"
                style={input}
                onChange={e =>
                  validate('customer_contact_email')(e) &&
                  onChange('customer_contact_email', e)
                }
                placeholder="Email Address"
                type="email"
                value={engagement.customer_contact_email || ''}
                data-cy={'customer_contact_email'}
              />
            </InputGroup>
          </FormGroup>
        </Form>
      </EditModalTemplate>
    </Modal>
  );
}
