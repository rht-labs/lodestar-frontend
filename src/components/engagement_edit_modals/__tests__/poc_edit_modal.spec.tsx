import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { PointOfContactEditModal } from '../point_of_contact_edit_modal';
import { Engagement } from '../../../schemas/engagement';
import { EngagementFormConfig } from '../../../schemas/engagement_config';
import MockDate from 'mockdate';
import { ValidationContext } from '../../../context/validation_context/validation_context';
import { TestStateWrapper } from '../../../common/test_state_wrapper';

describe('Point of Contact edit modal', () => {
  test('matches snapshot', () => {
    MockDate.set(new Date(2020, 8, 3));
    expect(
      render(
        <PointOfContactEditModal
          onSave={() => {}}
          formOptions={EngagementFormConfig.fromFake()}
          isOpen={true}
          engagement={Engagement.fromFake(true)}
          onChange={() => {}}
        />
      )
    ).toMatchSnapshot();
    MockDate.reset();
  });

  test('When clicking the save button, the onSave method is called', async () => {
    const onSave = jest.fn();
    const { getByTestId } = render(
      <PointOfContactEditModal
        onSave={onSave}
        engagement={Engagement.fromFake(true)}
        formOptions={{}}
        isOpen={true}
        onChange={() => {}}
      />
    );
    await fireEvent.click(getByTestId('poc-edit-save'));
    expect(onSave).toHaveBeenCalled();
  });
  test('When editing a field, the onChange method is called', async () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <PointOfContactEditModal
        onSave={() => {}}
        engagement={Engagement.fromFake(true)}
        formOptions={{}}
        isOpen={true}
        onChange={onChange}
      />
    );
    const textField = await getByTestId('el-email');
    await fireEvent.change(textField, {
      target: { value: 'bot' },
    });
    expect(onChange).toHaveBeenCalled();
  });
  test('Displays validation errors', async () => {
    const { findAllByText } = render(
      <TestStateWrapper>
        <ValidationContext.Provider
          value={{
            clearValidationResults: () => {},
            validationResults: {},
            getValidationResult: () => ['Incorrect input'],
          }}
        >
          <PointOfContactEditModal
            onSave={() => {}}
            engagement={Engagement.fromFake(true)}
            formOptions={{}}
            isOpen={true}
            onChange={() => {}}
          />
        </ValidationContext.Provider>
      </TestStateWrapper>
    );
    expect(await findAllByText('Incorrect input')).toHaveLength(3);
  });
  test('Modifying the email addresses calls validate', async () => {
    const validationSpy = jest.fn();
    const { getByTestId } = render(
      <TestStateWrapper>
        <ValidationContext.Provider
          value={{
            clearValidationResults: () => {},
            validationResults: {},
            getValidationResult: () => ['Incorrect input'],
            validate: () => validationSpy,
          }}
        >
          <PointOfContactEditModal
            onSave={() => {}}
            engagement={Engagement.fromFake(true)}
            formOptions={{}}
            isOpen={true}
            onChange={() => {}}
          />
        </ValidationContext.Provider>
      </TestStateWrapper>
    );
    await fireEvent.change(getByTestId('el-email'), {
      target: { value: 'bot' },
    });
    await fireEvent.change(getByTestId('tech-email'), {
      target: { value: 'bot' },
    });
    await fireEvent.change(getByTestId('customer-email'), {
      target: { value: 'bot' },
    });
    expect(validationSpy).toHaveBeenCalledTimes(3);
  });
});
