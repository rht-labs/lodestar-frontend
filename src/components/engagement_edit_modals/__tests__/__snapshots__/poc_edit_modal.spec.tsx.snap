// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point of Contact edit modal matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="pf-c-backdrop__open"
  >
    <div
      aria-hidden="true"
    />
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-labelledby="pf-modal-part-1"
            aria-modal="true"
            class="pf-c-modal-box pf-m-sm"
            data-ouia-component-id="0"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-0"
            role="dialog"
          >
            <button
              aria-disabled="false"
              aria-label="Close"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  transform=""
                />
              </svg>
            </button>
            <header
              class="pf-c-modal-box__header"
            >
              <h1
                class="pf-c-modal-box__title pf-c-modal-box__title"
                id="pf-modal-part-1"
              >
                Points of Contact
              </h1>
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
            >
              <section
                class="pf-c-page__main-section"
              >
                <form
                  class="pf-c-form"
                  novalidate=""
                >
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="engagement_lead"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          Labs EL
                        </span>
                        <span
                          aria-hidden="true"
                          class="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <div
                        class="pf-c-input-group"
                      >
                        <label
                          class="pf-c-input-group__text"
                          for="engagement-lead"
                          style="background-color: rgb(237, 237, 237);"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          >
                            <path
                              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="Engagement lead name"
                          class="pf-c-form-control"
                          data-cy="engagement_lead_name"
                          disabled=""
                          id="name"
                          name="engagement-lead-name"
                          placeholder="Full Name"
                          style="background-color: rgb(237, 237, 237);"
                          type="text"
                          value="Alice Doe"
                        />
                        <label
                          class="pf-c-input-group__text"
                          for="engagement-lead-email"
                          style="background-color: rgb(237, 237, 237);"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="engagement lead email"
                          class="pf-c-form-control"
                          data-cy="engagement_lead_email"
                          data-testid="el-email"
                          disabled=""
                          id="email"
                          name="engagement-lead-email"
                          placeholder="Email Address"
                          style="background-color: rgb(237, 237, 237);"
                          type="email"
                          value="alice@doe.com"
                        />
                      </div>
                      <div
                        aria-live="polite"
                        class="pf-c-form__helper-text"
                        id="engagement_lead-helper"
                      >
                        Who is the Engagement Lead?
                      </div>
                    </div>
                  </div>
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="technical_lead"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          Labs Technical Lead
                        </span>
                        <span
                          aria-hidden="true"
                          class="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <div
                        class="pf-c-input-group"
                        label="Labs Tech Lead"
                      >
                        <label
                          class="pf-c-input-group__text"
                          for="tech-lead-name"
                          style="background-color: rgb(237, 237, 237);"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          >
                            <path
                              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="Tech lead name"
                          class="pf-c-form-control"
                          data-cy="tech_lead_name"
                          disabled=""
                          id="tech-lead-name"
                          name="tech-lead-name"
                          placeholder="Full Name"
                          style="background-color: rgb(237, 237, 237);"
                          type="text"
                          value="Eve Doe"
                        />
                        <label
                          class="pf-c-input-group__text"
                          for="tech-lead-email"
                          style="background-color: rgb(237, 237, 237);"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="tech lead email"
                          class="pf-c-form-control"
                          data-cy="tech_lead_email"
                          data-testid="tech-email"
                          disabled=""
                          id="tech-lead-email"
                          name="tech-lead-email"
                          placeholder="Email Address"
                          style="background-color: rgb(237, 237, 237);"
                          type="email"
                          value="eve@doe.com"
                        />
                      </div>
                      <div
                        aria-live="polite"
                        class="pf-c-form__helper-text"
                        id="technical_lead-helper"
                      >
                        Who is the Tech Lead?
                      </div>
                    </div>
                  </div>
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="customer-contact"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          Customer Contact
                        </span>
                        <span
                          aria-hidden="true"
                          class="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <div
                        class="pf-c-input-group"
                        label="Customer Contact"
                      >
                        <label
                          class="pf-c-input-group__text"
                          for="customer-contact-name"
                          style="background-color: rgb(237, 237, 237);"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          >
                            <path
                              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="Customer contact name"
                          class="pf-c-form-control"
                          data-cy="customer_contact_name"
                          disabled=""
                          id="customer-contact-name"
                          name="customer-contact-name"
                          placeholder="Full Name"
                          style="background-color: rgb(237, 237, 237);"
                          type="text"
                          value="Bob Doe"
                        />
                        <label
                          class="pf-c-input-group__text"
                          for="customer-contact-email"
                          style="background-color: rgb(237, 237, 237);"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="customer contact email"
                          class="pf-c-form-control"
                          data-cy="customer_contact_email"
                          data-testid="customer-email"
                          disabled=""
                          id="customer-contact-email"
                          name="customer-contact-email"
                          placeholder="Email Address"
                          style="background-color: rgb(237, 237, 237);"
                          type="email"
                          value="bob@doe.com"
                        />
                      </div>
                      <div
                        aria-live="polite"
                        class="pf-c-form__helper-text"
                        id="customer-contact-helper"
                      >
                        Who is the point person for the project?
                      </div>
                    </div>
                  </div>
                </form>
              </section>
              <section
                class="pf-c-page__main-section"
              >
                <div
                  style="margin-top: 1rem;"
                >
                  <div
                    class="pf-l-flex pf-m-justify-content-flex-end"
                  >
                    <div
                      class=""
                    >
                      <div>
                        <button
                          aria-disabled="false"
                          class="pf-c-button pf-m-primary"
                          data-cy="save_point_of_contact"
                          data-ouia-component-id="2"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe="true"
                          data-testid="poc-edit-save"
                          type="button"
                        >
                          Save
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
